cmd/pomerium/main.go

```diff
@@ -32,7 +33,9 @@ func main() {
 	root.AddCommand(zero_cmd.BuildRootCmd())
 	root.PersistentFlags().StringVar(&configFile, "config", "", "Specify configuration file location")
 	log.SetLevel(zerolog.InfoLevel)
-	ctx := trace.NewContext(context.Background(), trace.NewSyncClient(nil))
+
+	sigCtx := contextutil.SetupSignals(context.Background())
```

Comment:
We have very similar code in `internal/zero/cmd/run.go`. Could we swap that out to use this new function?

Reply:
It looks like the zero_cmd already accepts the context constructed here:
```go
    sigCtx := contextutil.SetupSignals(context.Background())
	ctx := trace.NewContext(sigCtx, trace.NewSyncClient(nil))
	defer func() {
		if err := trace.ShutdownContext(ctx); err != nil {
			log.Error().Err(err).Send()
		}
	}()
	runFn := run
	if zero_cmd.IsManagedMode(configFile) {
		runFn = zero_cmd.Run
	}
	root.RunE = func(_ *cobra.Command, _ []string) error {
		defer log.Ctx(ctx).Info().Msg("cmd/pomerium: exiting")
		return runFn(ctx, configFile)
	}

```
so I'll just remove the `withInterrupt` wrapper in Zero